src/accelgyrotemp_ros_publisher.cpp

```diff
@@ -50,6 +50,9 @@ void AccelGyroTempRosPublisher::run()
     RCLCPP_INFO(rclcpp::get_logger("rclcpp_accelgyrotemp"), "startThread: '%d'", this_id);
 
     bool success = false;
+    bool firstData = true;
```

Comment:
also in this case firstData does not have much meaning. let's change its name and it logic:
new name: bufferedDataEnabled = false;
and if it is false call the API to enable it: enableBufferedDataOutput and set it to true if call is successful 
please apply the same logic to the other publisher